internal/controller/kubeconfig_provider_test.go

```diff
@@ -0,0 +1,129 @@
+//go:build integration
+
+package controller
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kuadrant/dns-operator/api/v1alpha1"
+)
+
+var _ = Describe("Kubeconfig Provider", Labels{"multicluster"}, func() {
+
+	var (
+		secondaryClusterSecret *v1.Secret
+
+		// buffer containing all the log entries added during the current spec execution
+		// can be used in conjunction with `gbytes.Say` matcher to check log entries see https://onsi.github.io/gomega/#codegbytescode-testing-streaming-buffers
+		logBuffer *gbytes.Buffer
+	)
+
+	BeforeEach(func() {
+		logBuffer = gbytes.NewBuffer()
+		GinkgoWriter.TeeTo(logBuffer)
+
+		secondaryClusterSecret = &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "secondary-cluster-1",
+				Namespace: testDefaultClusterSecretNamespace,
+				Labels: map[string]string{
+					testDefaultClusterSecretLabel: "true",
+				},
+			},
+			StringData: map[string]string{
+				"kubeconfig": string(secondaryKubeconfig),
+			},
+		}
+	})
+
+	AfterEach(func() {
+		if secondaryClusterSecret != nil {
+			Expect(client.IgnoreNotFound(primaryK8sClient.Delete(ctx, secondaryClusterSecret))).To(Succeed())
+		}
+		GinkgoWriter.ClearTeeWriters()
+	})
+
+	It("successfully creates a new cluster from valid kubeconfig secret", Labels{"primary"}, func(ctx SpecContext) {
+		createClusterKubeconfigSecret(primaryK8sClient, secondaryClusterSecret, logBuffer)
+	})
+
+	It("successfully deletes cluster", Labels{"primary"}, func(ctx SpecContext) {
```

Comment:
same here and line 65 too.

Reply:
changing it to "successfully removes connection to remote cluster"